src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/NMethod.java

```diff
@@ -416,10 +425,10 @@ public Map<sun.jvm.hotspot.debugger.Address, PCDesc> getSafepoints() {
   // FIXME: add isPatchableAt()
 
   /** Support for code generation. Only here for proof-of-concept. */
-  public static int getEntryPointOffset()            { return (int) entryPointField.getOffset();            }
-  public static int getVerifiedEntryPointOffset()    { return (int) verifiedEntryPointField.getOffset();    }
-  public static int getOSREntryPointOffset()         { return (int) osrEntryPointField.getOffset();         }
-  public static int getEntryBCIOffset()              { return (int) entryBCIField.getOffset();              }
+  public int getEntryPointOffset()            { return (int) entryOffsetField.getValue(addr);        }
+  public int getVerifiedEntryPointOffset()    { return (int) verifiedEntryOffsetField.getValue(addr);}
+  public static int getOSREntryPointOffset()  { return (int) osrEntryPointField.getOffset();         }
+  public static int getEntryBCIOffset()       { return (int) entryBCIField.getOffset();              }
```

Comment:
getMethodOffset() missing (from 8329433), is it not used?

Reply:
We didn't remove CompiledMethod from hotspot. So _method field is defined in CompiledMethod, not NMethod, getMethodOffset() is in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/CompiledMethod.java